server/svix-server/src/worker.rs

```diff
@@ -47,6 +49,27 @@ fn generate_msg_headers(
         headers.insert("svix-signature", signatures_str);
     }
 
+    let configured_headers =
+    	// FIXME: Don't clone if possible
+        endpoint::Entity::secure_find_by_id(msg_task.app_id.clone(), msg_task.endpoint_id.clone())
+            .one(db)
+            .await
+            .unwrap_or_default()
+            .map(|model| model.headers)
+            .flatten();
```

Comment:
You already have the endpoint outside of this function, so no need to fetch it again, just pass it. :)